frontends/infinite-corridor/src/pages/user-lists/UserListDetailsPage.tsx

```diff
@@ -0,0 +1,85 @@
+import React from "react"
+import { BannerPage } from "ol-util"
+import { GridColumn, GridContainer } from "../../components/layout"
+import { useUserListItems, useUserList } from "../../api/learning-resources"
+import Container from "@mui/material/Container"
+import EditIcon from "@mui/icons-material/Edit"
+import Grid from "@mui/material/Grid"
+import Button from "@mui/material/Button"
+import { LearningResourceCard } from "ol-search-ui"
+import { EditListDialog, useEditingDialog } from "./ManageListDialogs"
+import { imgConfigs } from "../../util/constants"
+import { useParams } from "react-router"
+import { useActivateResourceDrawer } from "../LearningResourceDrawer"
+
+type RouteParams = {
+  id: string
+}
+
+const UserListsDetailsPage: React.FC = () => {
+  const id = Number(useParams<RouteParams>().id)
+  const userListQuery = useUserList(id)
+  const itemsDataQuery = useUserListItems(id)
+  const activateResource = useActivateResourceDrawer()
+  const editing = useEditingDialog()
+
+  return (
+    <BannerPage
+      src="/static/images/course_search_banner.png"
+      alt=""
+      compactOnMobile
+    >
+      <Container maxWidth="sm" className="userlist-page">
+        <GridContainer>
+          <GridColumn variant="single-full">
+            {userListQuery.data && (
+              <Grid container className="ic-list-header">
+                <Grid item xs={9}>
+                  <h1>{userListQuery.data.title}</h1>
+                </Grid>
+                <Grid item xs={3} className="ic-centered-right">
+                  <Button
+                    color="secondary"
+                    startIcon={<EditIcon />}
+                    onClick={() => editing.handleStart(userListQuery.data)}
+                  >
+                    Edit
+                  </Button>
+                </Grid>
+              </Grid>
+            )}
+            {itemsDataQuery.isLoading && <p>Loading...</p>}
+            {itemsDataQuery.data &&
+              (itemsDataQuery.data.results.length === 0 ? (
+                <p className="empty-message">
+                  There are no items in this list yet.
+                </p>
+              ) : (
+                <ul className="ic-card-row-list">
+                  {itemsDataQuery.data.results.map(list => {
+                    return (
+                      <li key={list.id}>
+                        <LearningResourceCard
+                          variant="row-reverse"
+                          className="ic-resource-card"
+                          resource={list.content_data}
+                          imgConfig={imgConfigs["row-reverse"]}
+                          onActivate={activateResource}
+                        />
+                      </li>
+                    )
+                  })}
+                </ul>
+              ))}
```

Comment:
You mentioned this and `FavoritesPage` don't make sense to be the same component, but I think you could probably pull this bit out into its own component since it's the same on both pages except for the `empty-message` text.